lib/utils/__tests__/report.test.mjs

```diff
@@ -550,3 +563,304 @@ test('without node nor line', () => {
 		'The "foo" rule failed to pass either a node or a line number to the `report()` function.',
 	);
 });
+
+test('without node', () => {
+	const v = {
+		ruleName: 'foo',
+		result: {
+			warn: jest.fn(),
+			stylelint: resultStylelint(),
+		},
+		message: 'bar',
+	};
+
+	try {
+		report(v);
+	} catch {
+		// noop
+	}
+
+	expect(emitWarning).toHaveBeenCalledWith(
+		expect.stringMatching(
+			/Omitting the `node` argument in the `utils.report\(\)` function is deprecated \("foo"\)\./,
+		),
+		expect.objectContaining({
+			code: 'stylelint:007',
+			detail: 'Please pass a `node` argument in the `utils.report()` function of "foo".',
+			type: 'DeprecationWarning',
+		}),
+	);
+});
+
+test('with quietDeprecationWarnings mode on and without node', () => {
+	const v = {
+		ruleName: 'foo',
+		result: {
+			warn: jest.fn(),
+			stylelint: resultStylelint({ quietDeprecationWarnings: true }),
+		},
+		message: 'bar',
+	};
+
+	try {
+		report(v);
+	} catch {
+		// noop
+	}
+
+	expect(emitWarning).toHaveBeenCalledTimes(0);
+});
+
+test('with line', () => {
+	const v = {
+		ruleName: 'foo',
+		result: {
+			warn: jest.fn(),
+			stylelint: resultStylelint(),
+		},
+		message: 'bar',
+		node: {
+			rangeBy: defaultRangeBy,
+		},
+		line: 1,
+	};
+
+	report(v);
+
+	expect(emitWarning).toHaveBeenCalledWith(
+		expect.stringMatching(
+			/Providing the `line` argument in the `utils.report\(\)` function is deprecated \("foo"\)\./,
+		),
+		expect.objectContaining({
+			code: 'stylelint:007',
+			detail:
+				'Please pass both `index` and `endIndex` as arguments in the `utils.report()` function of "foo" instead.',
```

Comment:
question

Maybe _to the_ instead of _in the_?
e.g. pass x as y to the z

Reply:
I think `in` is also fine.
Let's keep it as-is, as this is only temporary text.